File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/FavoriteApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class FavoriteApi {
    private final HttpClient client;
    
    public FavoriteApi(HttpClient client) {
        this.client = client;
    }

    /** Update an existing favorite */
    public PlusApiResultPlusFavoriteVO update(PlusFavoriteForm body) throws Exception {
        return (PlusApiResultPlusFavoriteVO) client.put(ApiPaths.backendPath("/favorite"), body);
    }

    /** Create a new favorite */
    public PlusApiResultPlusFavoriteVO create(PlusFavoriteForm body) throws Exception {
        return (PlusApiResultPlusFavoriteVO) client.post(ApiPaths.backendPath("/favorite"), body);
    }

    /** Get favorites by page */
    public PlusApiResultPagePlusFavoriteVO listByPage(QueryListForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPagePlusFavoriteVO) client.post(ApiPaths.backendPath("/favorite/list"), body, params);
    }

    /** Get all favorites */
    public PlusApiResultListPlusFavoriteVO listAllEntities(QueryListForm body) throws Exception {
        return (PlusApiResultListPlusFavoriteVO) client.post(ApiPaths.backendPath("/favorite/list/all"), body);
    }

    /** Get a favorite by ID */
    public PlusApiResultPlusFavoriteVO getById(String id) throws Exception {
        return (PlusApiResultPlusFavoriteVO) client.get(ApiPaths.backendPath("/favorite/" + id + ""));
    }

    /** Delete a favorite */
    public PlusApiResultBoolean delete(String id) throws Exception {
        return (PlusApiResultBoolean) client.delete(ApiPaths.backendPath("/favorite/" + id + ""));
    }
}

File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/api/FavoriteApi.kt
package com.sdkwork.backend.api

import com.sdkwork.backend.*
import com.sdkwork.backend.http.HttpClient

class FavoriteApi(private val client: HttpClient) {

    /** Update an existing favorite */
    suspend fun update(body: PlusFavoriteForm): PlusApiResultPlusFavoriteVO? {
        return client.put(ApiPaths.backendPath("/favorite"), body) as? PlusApiResultPlusFavoriteVO
    }

    /** Create a new favorite */
    suspend fun create(body: PlusFavoriteForm): PlusApiResultPlusFavoriteVO? {
        return client.post(ApiPaths.backendPath("/favorite"), body) as? PlusApiResultPlusFavoriteVO
    }

    /** Get favorites by page */
    suspend fun listByPage(body: QueryListForm? = null, params: Map<String, Any>? = null): PlusApiResultPagePlusFavoriteVO? {
        return client.post(ApiPaths.backendPath("/favorite/list"), body, params) as? PlusApiResultPagePlusFavoriteVO
    }

    /** Get all favorites */
    suspend fun listAllEntities(body: QueryListForm? = null): PlusApiResultListPlusFavoriteVO? {
        return client.post(ApiPaths.backendPath("/favorite/list/all"), body) as? PlusApiResultListPlusFavoriteVO
    }

    /** Get a favorite by ID */
    suspend fun getById(id: String): PlusApiResultPlusFavoriteVO? {
        return client.get(ApiPaths.backendPath("/favorite/$id")) as? PlusApiResultPlusFavoriteVO
    }

    /** Delete a favorite */
    suspend fun delete(id: String): PlusApiResultBoolean? {
        return client.delete(ApiPaths.backendPath("/favorite/$id")) as? PlusApiResultBoolean
    }
}
